test/parallel/test-http-server-optimize-empty-requests.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const net = require('net');
+
+let reqs = 0;
+const server = http.createServer({
+  optimizeEmptyRequests: true
+}, (req, res) => {
+  reqs++;
+  req.on('data', common.mustNotCall());
+  req.on('end', common.mustNotCall());
+
+  assert.strictEqual(req._dumped, true);
+  assert.strictEqual(req._readableState.ended, true);
```

Comment:
I think it would be better to assert on the public API surface `req.readableEnded`.